dotCMS/src/main/java/com/dotmarketing/common/db/DotConnect.java

```diff
@@ -1251,4 +1257,14 @@ private static void setTimestampWithTimezone(PreparedStatement statement, int pa
         }
     }
 
+    /**
+     *  Number of rows to return, if zero is being ignore.
+     *  for instance, if using a forward cursor to fetch heavy rows, you can set to 1 (it will be slower but memory resources will be happy)
+     *  The behaviour will depend on the driver
+     * */
+    public DotConnect setFetchSize(final int fetchSize) {
+
+        this.fetchSize = fetchSize;
```

Comment:
I think we should not have this here  - b/c it gives a false sense that you are doing something to avoid excessive memory consumption, when in fact all results are still being loaded into. memory.

Reply:
removed from DotConnect and replaced it use by plain raw JDBC